docs/resources/tutorials/differential-expression-analysis.md

```diff
@@ -0,0 +1,458 @@
+
+```bash title="dme_elev_samples.tsv"
+sample population temp
+SRR1576457 maine low
+SRR1576458 maine low
+SRR1576459 maine low
+SRR1576460 maine high
+SRR1576461 maine high
+SRR1576462 maine high
+SRR1576463 panama low
+SRR1576464 panama low
+SRR1576465 panama low
+SRR1576514 panama high
+SRR1576515 panama high
+SRR1576516 panama high
+```
+
+### 1. Expression analysis in R: preliminaries
+The differential expression analysis we demonstrate requires a number of R packages, which can be installed as follows:
+
+```R
+install.packages("tidyverse")
+
+if (!require("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+
+BiocManager::install("limma","edgeR")
+```
+
+`Tidyverse` is a data science packages that facilitates the manipulation of tabular data, and plotting visualizations. `edgeR` and `limma` are bioinformatics packages for analyzing bulk RNA-seq data.
+
+### 2. Load gene-level abundance data
+The first two columns of `salmon.merged.gene_counts.tsv` are "gene_id", the original gene id in the annotation, and "gene_name", the gene symbol. There may be cases in which a gene symbol is not unique among gene ids, such that it is useful to concatenate these two columns into a new label, which we do here. To manipulate the data into a matrix format that limma expects, we also need to remove the id and name information from the data table, and set the new concatenated label as row names. We do basic table manipulation by loading the table as a tibble via tidyverse, but then, since tidyverse doesn't accomodate row names, convert back to a data frame.
+
+```R
+expression_data <-read_table("salmon.merged.gene_counts.tsv") %>%
+                  mutate(gene_id_symbol=paste(gene_id,gene_name,sep="_")) %>%
+                  select(!c(gene_id,gene_name)) %>%
+                  relocate(gene_id_symbol)
+
+new_ids=expression_data$gene_id_symbol
+expression_data <- expression_data %>% select(!c(gene_id_symbol))
+expression_data <-as.data.frame(expression_data)
+row.names(expression_data)=new_ids
+```
+
+### 3. Load sample information
+We not only load the sample info, but also create a new variable, *pop_temp*, that is the concatenation of the two categorical variables. We will use this concatenated variable to filter out lowly expressed genes, to insure that retained genes have sufficient counts for performing data analysis across samples consisting of each combination of temperature and population.
+
+```R
+sample_info <- read.table("expression_data/dme_elev_samples.tsv",header = TRUE, stringsAsFactors=FALSE)
+sample_info$pop_temp<-paste(sample_info$population,sample_info$temp,sep="_")
```

Comment:
Is this really just for filtering? i guess maybe if you don't do any complicated contrasts in later steps. but if you don't use this for anything but defining the group size for filtering it probably is not needed?